src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -1413,4 +1413,69 @@ BOOST_AUTO_TEST_CASE(BRNNTrainReturnObjective)
   BOOST_REQUIRE_EQUAL(std::isfinite(objVal), true);
 }
 
+/**
+ * Test that RNN::Train() does not give an error for large rho.
+ */
+BOOST_AUTO_TEST_CASE(LargeRhoValueRnnTest)
+{
+  // Setting rho value greater than sequence length which is 17.
+  const size_t rho = 100;
+  const size_t hiddenSize = 128;
+  const size_t numLetters = 256;
+  using MatType = arma::cube;
+  std::vector<std::string> trainingData;
+  trainingData.push_back(std::string("THIS IS THE INPUT 0"));
+  trainingData.push_back(std::string("THIS IS THE INPUT 1"));
+  trainingData.push_back(std::string("THIS IS THE INPUT 2"));
+
+  RNN<> model(rho);
+  model.Add<IdentityLayer<>>();
+  model.Add<LSTM<>>(numLetters, hiddenSize, rho);
+  model.Add<Dropout<>>(0.1);
+  model.Add<Linear<>>(hiddenSize, numLetters);
+
+  const auto makeInput = [numLetters](const char *line) -> MatType
+  {
+    const auto strLen = strlen(line);
+    // Rows: number of dimensions.
+    // Cols: number of sequences/points.
+    // Slices: number of steps in sequences.
+    MatType result(numLetters, 1, strLen, fill::zeros);
+    for (size_t i = 0; i < strLen; ++i)
+    {
+      const auto letter = line[i];
+      result.at(static_cast<uword>(letter), 0, i) = 1.0;
+    }
+    return result;
+  };
+
+  const auto makeTarget = [] (const char *line) -> MatType
+  {
+    const auto strLen = strlen(line);
+    // Responses for NegativeLogLikelihood should be
+    // non-one-hot-encoded class IDs (from 1 to num_classes).
+    MatType result(1, 1, strLen, fill::zeros);
+    // The response is the *next* letter in the sequence.
+    for (size_t i = 0; i < strLen - 1; ++i)
+    {
+      const auto letter = line[i + 1];
+      result.at(0, 0, i) = static_cast<uword>(letter) + 1.0;
+    }
+    // The final response is empty, so we set it to class 0.
+    result.at(0, 0, strLen - 1) = 1.0;
+    return result;
+  };
+
+  std::vector<MatType> inputs(trainingData.size());
+  std::vector<MatType> targets(trainingData.size());
+  for (size_t i = 0; i < trainingData.size(); ++i)
```

Comment:
May be we can use range based for loop here 

```
For ( auto sentence : trainingData)
{
    inputs[i] = makeInput(sentence.c_str());
    targets[i] = makeTarget(sentence.c_str());
}
```

Reply:
Hey, @jeffin143 thanks for helping I think may be we should keep the loop as it is because we need to assign a new variable.
`i = position of sentence` .